### PR TITLE
Fix version comparison

### LINKED ISSUE #334: Version Comparison is broken when galaxy version >= 4.10.0
Error: https://github.com/ansible/galaxy_ng/actions/runs/6892485007/job/18749880822?pr=1981

"msg": "This module requires private automation hub version 4.6 or later. Your version is 4.10.0"

This happens because we are bumping galaxy_ng version to "4.10.0" and the comparison on modules are using string comparison.

https://github.com/search?q=repo%3Aansible%2Fgalaxy_collection%20%22this%20module%20requires%22&type=code

## Proposal

Use one of:

Needs packaging as a requirement (is that already in Ansible)?
```python
>>> # pip install packaging
>>> from packaging import version
>>> version.parse("2.3.1") < version.parse("10.1.2")
True
>>> version.parse("1.3.a4") < version.parse("10.1.2")
True
>>> isinstance(version.parse("1.3.a4"), version.Version)
True
>>> isinstance(version.parse("1.3.xy123"), version.LegacyVersion)
True
```

OR

```python
>>> from pkg_resources import parse_version
>>> parse_version('1.9.a.dev') == parse_version('1.9a0dev')
True
>>> parse_version('2.1-rc2') < parse_version('2.1')
True
>>> parse_version('0.6a9dev-r41475') < parse_version('0.6a9')
True
```

### PR DESCRIPTION
**Pull request by rochacbruno**
Compara versions using ansible.module_utils.compat.version.LooseVersion

closes #334

### CODE REVIEW

**Comment by rochacbruno** [comment]
packaging is added by ansible https://github.com/ansible/ansible/blob/devel/requirements.txt#L9

Not sure why the CI is failing, do we need to include it elsewhere?

**Review by sean-m-sullivan — APPROVED** [review]
Clean, thanks!

**Comment by rochacbruno** [comment]
@sean-m-sullivan CI is failing because the packaging dependency cannot be found

**Comment by sean-m-sullivan** [comment]
it looks like this dependency is not part of the ansible install. need it in requirements.txt in the base directory.

**Comment by rochacbruno** [comment]
ALternativelly we can create our own implementation of the class `Version` and not depend on external library.

```python
class Version:
    def __init__(self, version_str):
        components = list(map(int, version_str.split('.')))
        self.major = components[0]
        self.minor = components[1] if len(components) > 1 else 0
        self.patch = components[2] if len(components) > 2 else 0

    def __str__(self):
        return f"{self.major}.{self.minor}.{self.patch}"

    def __eq__(self, other):
        if not isinstance(other, Version):
            return NotImplemented
        return (self.major, self.minor, self.patch) == (other.major, other.minor, other.patch)

    def __lt__(self, other):
        if not isinstance(other, Version):
            return NotImplemented
        return (self.major, self.minor, self.patch) < (other.major, other.minor, other.patch)

    def __le__(self, other):
        if not isinstance(other, Version):
            return NotImplemented
        return (self < other) or (self == other)

    def __gt__(self, other):
        if not isinstance(other, Version):
            return NotImplemented
        return not (self <= other)

    def __ge__(self, other):
        if not isinstance(other, Version):
            return NotImplemented
        return not (self < other)

    def __ne__(self, other):
        if not isinstance(other, Version):
            return NotImplemented
        return not (self == other)

# Example usage:
version1 = Version("4.2")
version2 = Version("5.1.3")

print(version1 < version2)  # Output: True
print(version1 >= version2)  # Output: False

```

File: plugins/modules/ah_ee_image.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 
 from __future__ import absolute_import, division, print_function
+from ansible.module_utils.compat.version import LooseVersion as Version
```
Comment (Tompage1994) [inline plugins/modules/ah_ee_image.py]: sanity check is stating that this needs to be with the imports further down below the documentation segment

Comment (Tompage1994) [reply plugins/modules/ah_ee_image.py]: Plus the same for all the other modules

**Review by Tompage1994 — CHANGES_REQUESTED** [review]

**Comment by rochacbruno** [comment]
Trying a simpler implementation on https://github.com/ansible/galaxy_collection/pull/337

**Comment by rochacbruno** [comment]
Closing in favor of #337

**Pull request closed without merging**